torch/_dynamo/variables/builder.py

```diff
@@ -1848,7 +1848,7 @@ def __call__(self, tx, value) -> VariableTracker:
             return SourcelessBuilder.wrap_constant_literal(value)
         elif is_builtin_callable(value):
             return BuiltinVariable(value)
-        elif is_allowed(value):
+        elif is_allowed(value) or isinstance(value, (torch.device, torch.dtype)):
             if is_user_defined_allowed(value):
                 self.tx.output.has_user_defined_allowed_in_graph = True
             return TorchVariable(value)
```

Comment:
I don't think `torch.device/torch.dtype` are torch variables. 

They are `ConstantVariable`s

You need to change the definition of `is_literal`

Reply:
The existing tests for torch.device show it as a TorchVariable 

https://github.com/pytorch/pytorch/blob/85dedeab654366400028e83cf34c8daad8a6afb9/test/dynamo/test_misc.py#L7505-L7507

Is torch.device a TorchVariable but torch.dtype should be a constant?

Reply:
Hmm that's weird, as variablebuilder seems to wrap it as a constantvariable. I guess both are technically fine, but changing is_literal can get rid of the special-casing here.

Reply:
https://github.com/pytorch/pytorch/blob/919c9b713e65aa1489123259a0ce7e79322d4d9d/torch/_dynamo/variables/builder.py#L302

Reply:
```torch.device/torch.dtype``` should be ```ConstantVariable```.